internal/services/steelthreadtesting/steelthread_test.go

```diff
@@ -0,0 +1,127 @@
+//go:build steelthread
+// +build steelthread
+
+package steelthreadtesting
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	yamlv3 "gopkg.in/yaml.v3"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/authzed/spicedb/internal/datastore/memdb"
+	"github.com/authzed/spicedb/internal/dispatch/graph"
+	"github.com/authzed/spicedb/internal/grpchelpers"
+	"github.com/authzed/spicedb/internal/middleware/consistency"
+	datastoremw "github.com/authzed/spicedb/internal/middleware/datastore"
+	v1svc "github.com/authzed/spicedb/internal/services/v1"
+	"github.com/authzed/spicedb/pkg/validationfile"
+	"github.com/dustin/go-humanize"
+	"github.com/gosimple/slug"
+	"github.com/stretchr/testify/require"
+)
+
+const defaultConnBufferSize = humanize.MiByte
+
+func TestSteelThreads(t *testing.T) {
+	for _, tc := range steelThreadTestCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+			defer cancel()
+
+			// Create a new datastore.
+			ds, err := memdb.NewMemdbDatastore(0, 0*time.Second, memdb.DisableGC)
+			require.NoError(t, err)
+
+			ctx = datastoremw.ContextWithDatastore(ctx, ds)
+
+			// Load in the data.
+			_, _, err = validationfile.PopulateFromFiles(ctx, ds, []string{"testdata/" + tc.datafile})
+			require.NoError(t, err)
+
+			// Create a new server.
+			listener := bufconn.Listen(defaultConnBufferSize)
+
+			s := grpc.NewServer(
```

Comment:
there's already a helper for spinning up servers like this in `internal/testserver`

Reply:
The testserver takes datastore init as an argument, we use it for datastore tests as well

Reply:
Okay, I'll switch to that